business/definitionService.js

```diff
@@ -38,15 +38,27 @@ const currentSchema = '1.6.1'
 
 const weights = { declared: 30, discovered: 25, consistency: 15, spdx: 15, texts: 15, date: 30, source: 70 }
 
+function stringHash(src) {
+  var hash = 0, i, chr
+  if (src.length === 0) return hash
+  for (i = 0; i < src.length; i++) {
+    chr = src.charCodeAt(i)
+    hash = ((hash << 5) - hash) + chr
+    hash |= 0
+  }
+  return hash
```

Comment:
Is it possible to use node crypto library instead of custom hash function?
https://nodejs.org/docs/latest-v10.x/api/crypto.html#crypto_crypto_createhash_algorithm_options

Reply:
Hi! It is possible, but unnecessarily expensive (we do not care about the same kind of hash algo properties that you look for when it’s security related; this is only being use to reduce the payload).

Reply:
For further illustration the article at https://searchsqlserver.techtarget.com/definition/hashing ends with the following assertion.
> A hash function that works well for database storage and retrieval, however, might not work as for cryptographic or error-checking purposes.